DependencyInjection/Compiler/CsrfExtensionPass.php

```diff
@@ -29,6 +29,17 @@ public function process(ContainerBuilder $container)
                 $tokenStorageReference = new Reference('security.context');
             }
             $definition->replaceArgument(0, $tokenStorageReference);
+
+            // Symfony <2.8 BC
+            $typeExtensionTag = $definition->getTag('form.type_extension');
+            if (method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')) {
+                $typeExtensionTag['extended-type'] = 'Symfony\Component\Form\Extension\Core\Type\FormType';
+            } else {
+                $typeExtensionTag['alias'] = 'form';
+            }
+            $definition->setTags(array_merge($definition->getTags(), array(
+                'form.type_extension' => $typeExtensionTag,
+            )));
```

Comment:
> hmm, even much simpler actually: set both alias and extended-type in the tag. the compiler pass in 2.8 will ignore alias if type-extension is set, and it will use the alias in 2.7

Oh, indeed.

In this case, would not be better to set both of them directly under the xml config file?